Client/Rewards/BraveRewards.swift

```diff
@@ -262,43 +256,96 @@ class BraveRewards: NSObject {
+    public var adsBuildChannel: Ads.BuildChannelInfo = .init()
+    public var isDebug: Bool?
+    public var overridenNumberOfSecondsBetweenReconcile: Int?
+    public var retryInterval: Int?
+
+    public static func current(
+      buildChannel: AppBuildChannel = AppConstants.buildChannel,
+      isDebugFlag: Bool? = Preferences.Rewards.debugFlagIsDebug.value,
+      retryInterval: Int? = Preferences.Rewards.debugFlagRetryInterval.value,
+      reconcileInterval: Int? = Preferences.Rewards.debugFlagReconcileInterval.value
+    ) -> Self {
+      var configuration: BraveRewards.Configuration
+      if !buildChannel.isPublic {
+        if let override = Preferences.Rewards.EnvironmentOverride(rawValue: Preferences.Rewards.environmentOverride.value), override != .none {
+          switch override {
+          case .dev:
+            configuration = .default
+          case .staging:
+            configuration = .staging
+          case .prod:
+            configuration = .production
+          default:
+            configuration = .staging
+          }
+        } else {
+          configuration = AppConstants.buildChannel == .debug ? .staging : .production
+        }
+        configuration.isDebug = isDebugFlag
+        configuration.retryInterval = retryInterval
+        configuration.overridenNumberOfSecondsBetweenReconcile = reconcileInterval
+      } else {
+        configuration = .production
+      }
+      return configuration
+    }
+    
     static var `default`: Configuration {
       .init(
         storageURL: FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!,
-        ledgerEnvironment: .development,
-        adsEnvironment: .staging
+        environment: .development
       )
     }
     static var staging: Configuration {
       var config = Configuration.default
-      config.ledgerEnvironment = .staging
-      config.adsEnvironment = .staging
+      config.environment = .staging
       return config
     }
     static var production: Configuration {
       var config = Configuration.default
-      config.ledgerEnvironment = .production
-      config.adsEnvironment = .production
+      config.environment = .production
       return config
     }
     static var testing: Configuration {
       .init(
         storageURL: URL(fileURLWithPath: NSTemporaryDirectory()),
-        ledgerEnvironment: .development,
-        adsEnvironment: .staging,
-        isTesting: true,
+        environment: .development,
         overridenNumberOfSecondsBetweenReconcile: 30,
         retryInterval: 30
       )
     }
+    
+    public var flags: String {
+      var flags: [String: String] = [:]
+      switch environment {
+      case .production:
+        flags["staging"] = "false"
+      case .staging:
+        flags["staging"] = "true"
+      case .development:
+        flags["development"] = "true"
+      }
+      if let debug = isDebug {
+        flags["debug"] = debug ? "true" : "false"
```

Comment:
Would `String(debug)` work here to cast bool to String?

Reply:
TIL there was an initializer on `String` that takes a `Bool`, yeah that would be better

Reply:
fixed in https://github.com/brave/brave-ios/pull/5960/commits/d16631c3cb38b8978f6c1d939b4ea15b7701ab43